frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/index.tsx

```diff
@@ -90,42 +90,68 @@ const BasicSelector: React.SFC<SpecCapabilityProps> = ({ value, capability }) =>
 );
 
 const BooleanSwitch: React.FC<SpecCapabilityProps> = (props) => {
+  const { model, obj, descriptor, onHandleError } = props;
   const convertedValue = !_.isNil(props.value) ? props.value : false;
   const [value, setValue] = React.useState(convertedValue);
   const [confirmed, setConfirmed] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState(null);
+
+  const errorCb = (err: Error): void => {
+    setConfirmed(false);
+    setValue(convertedValue);
+    setErrorMessage(err.message);
+    onHandleError(err.message);
+  };
 
-  const patchFor = (val: boolean) => [
-    { op: 'add', path: `/spec/${props.descriptor.path.replace('.', '/')}`, value: val },
-  ];
   const update = () => {
     setConfirmed(true);
-    return k8sPatch(props.model, props.obj, patchFor(value));
+    setErrorMessage(null);
+
+    const patchFor = (val: boolean) => [
+      { op: 'add', path: `/spec/${descriptor.path.replace(/\./g, '/')}`, value: val },
+    ];
+
+    let newObj = _.cloneDeep(obj);
+    newObj = _.set(newObj, `spec.${descriptor.path}`, value);
+
+    if (props.value) {
```

Comment:
We need some comments explaining why we're doing this.

Is this the right way to test this? I was expecting something like

```ts
if (_.has(obj, `spec.${descriptor.path}`)) {
```

Reply:
> We need some comments explaining why we're doing this.
> 
> Is this the right way to test this? I was expecting something like
> 
> ```ts
> if (_.has(obj, `spec.${descriptor.path}`)) {
> ```

@spadgett There is a similar check earlier in the flow. props.value returns undefined if the property value doesn't exist or unset. 

https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx#L469

https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx#L541

Reply:
That's not exactly the same because `descriptor.value` is a default/fallback value that's provided in the descriptor. So it's possible `props.value` is defined, but it's not set in the resource.

Reply:
> That's not exactly the same because `descriptor.value` is a default/fallback value that's provided in the descriptor. So it's possible `props.value` is defined, but it's not set in the resource.

@spadgett Oh I see. Will use lodash _.has as mentioned above to do the check. Thanks.